src/BenchmarkDotNet/Toolchains/NativeAot/Generator.cs

```diff
@@ -237,5 +262,55 @@ private void GenerateReflectionFile(ArtifactsPaths artifactsPaths)
         }
+
+        private string GetCurrentInstructionSet(Platform platform)
+            => string.Join(",", GetHostProcessInstructionSets(platform));
+
+        private static IEnumerable<string> GetHostProcessInstructionSets(Platform platform)
+        {
+            switch (platform)
+            {
+                case Platform.X86:
+                case Platform.X64:
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.X86Base")) yield return "base";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Sse")) yield return "sse";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Sse2")) yield return "sse2";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Sse3")) yield return "sse3";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Sse41")) yield return "sse4.1";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Sse42")) yield return "sse4.2";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Avx")) yield return "avx";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Avx2")) yield return "avx2";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Aes")) yield return "aes";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Bmi1")) yield return "bmi";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Bmi2")) yield return "bmi2";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Fma")) yield return "fma";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Lzcnt")) yield return "lzcnt";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Pclmulqdq")) yield return "pclmul";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.Popcnt")) yield return "popcnt";
+                    if (GetIsSupported("System.Runtime.Intrinsics.X86.AvxVnni")) yield return "avxvnni";
+                    break;
+                case Platform.Arm64:
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.ArmBase")) yield return "base";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.AdvSimd")) yield return "neon";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Aes")) yield return "aes";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Crc32")) yield return "crc";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Dp")) yield return "dotprod";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Rdm")) yield return "rdma";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Sha1")) yield return "sha1";
+                    if (GetIsSupported("System.Runtime.Intrinsics.Arm.Sha256")) yield return "sha2";
+                    // todo: handle "lse"
```

Comment:
@jkotas when can "lse" be enabled? https://github.com/dotnet/runtime/blob/ce61c09a5f6fc71d8f717d3fc4562f42171869a0/src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs#L741

Reply:
`lse` enables more efficient code generation for `Interlocked` APIs. It is not exposed via hardware intrinics managed APIs. `dczva` and `rcpc` are in the same category.

You would have to replicate what the unmanaged runtime does to auto-detect it.